pkg/controllers/dynakube/injection/reconciler.go

```diff
@@ -9,26 +9,32 @@ import (
 	"github.com/Dynatrace/dynatrace-operator/pkg/clients/dynatrace"
 	"github.com/Dynatrace/dynatrace-operator/pkg/controllers"
 	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/istio"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/processmoduleconfigsecret"
 	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/version"
 	"github.com/Dynatrace/dynatrace-operator/pkg/injection/namespace/ingestendpoint"
 	"github.com/Dynatrace/dynatrace-operator/pkg/injection/namespace/initgeneration"
 	"github.com/Dynatrace/dynatrace-operator/pkg/injection/namespace/mapper"
 	"github.com/Dynatrace/dynatrace-operator/pkg/util/timeprovider"
 	"github.com/spf13/afero"
+	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 type reconciler struct {
-	client            client.Client
-	apiReader         client.Reader
-	dynakube          *dynatracev1beta1.DynaKube
-	istioReconciler   istio.Reconciler
-	versionReconciler version.Reconciler
+	client              client.Client
+	apiReader           client.Reader
+	dynakube            *dynatracev1beta1.DynaKube
+	istioReconciler     istio.Reconciler
+	versionReconciler   version.Reconciler
+	pmcSecretreconciler processmoduleconfigsecret.Reconciler
```

Comment:
use the `interface` not the exact `struct`, so it can be mocked fully

- currently you can "mock" it by creating a `processmoduleconfigsecret.Reconciler` with fakeClients and stuff, but that is not necessarily what we want always
  - considering the `injection.Reconciler` as a "unit", it should only care that that given reconciler was called
  - you can have a bigger test where you would care how each reconciler actually interact, but that is a system/e2e test and not a unittest. (which you can still do if you use the `interface` instead of the `struct`)

Reply:
`controllers.Reconciler` is the `interface` I am talking about 😅

Reply:
if i understood it correctly I did it in https://github.com/Dynatrace/dynatrace-operator/pull/2777/commits/8d3c72599d3747501778e8602d73690edcee9bce :sweat_smile: